Week1/js-exercises/ex5-catWalk.js

```diff
@@ -10,4 +10,30 @@
  5. When the cat reaches the right - hand of the screen, restart them at the left hand side("0px").So they should keep walking from left to right across the screen, forever and ever.
  6. When the cat reaches the middle of the screen, replace the img with an image of a cat dancing(use this URL: https: //tenor.com/StFI.gif), keep it dancing for 5 seconds, and then replace the img with the original image and have it continue the walk.
  
-*/
\ No newline at end of file
+*/
+const catImg = document.querySelector('body img');
+let point = -300;
+app();
+
+function app() {
+
+  const walking = setInterval(catWalk, 5);
+
+  function catWalk() {
+    if (point > screen.width) {
+      point = -300;
+    } else if (point == (screen.width / 2) - 150) {
+      clearInterval(walking);
+      catImg.src = 'tenor.gif';
+      setTimeout(() => {
+        catImg.src = 'http://www.anniemation.com/clip_art/images/cat-walk.gif';
+        point++;
+        app();
```

Comment:
Are you doing a recursive function?

This works properly? 

It can be considered a bit confusing to read this code but if it works now then its fine

Reply:
Yes, It works.
And I don't know recursive function, I use this way by myself and really don't know why use this, maybe by my
unconscious. In Persian we say 'سعی و خطا'